jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerEntry.java

```diff
@@ -52,10 +63,21 @@
    * @param sourceFile the source file to add to the layer
    * @param extractionPath the path in the container file system corresponding to the {@code
    *     sourceFile}
+   * @param permissions the file permissions on the container. Use {@code null} to use defaults (644
+   *     for files, 755 for directories)
    */
-  public LayerEntry(Path sourceFile, AbsoluteUnixPath extractionPath) {
+  public LayerEntry(
+      Path sourceFile,
+      AbsoluteUnixPath extractionPath,
+      @Nullable Set<PosixFilePermission> permissions) {
     this.sourceFile = sourceFile;
     this.extractionPath = extractionPath;
+    if (permissions == null) {
+      this.permissions =
```

Comment:
I'm wondering if this logic might not be better put into the tar-building so that we can differentiate between files that had explicit permissions and those that didn't.  This would be important if we modify the Dockerfile generator to do a `RUN chmod xxx` for the specific files.

Reply:
That's true, I think I'll move it.